docs/src/tutorials/advanced_simulations.md

```diff
@@ -0,0 +1,215 @@
+
+### Monte Carlo simulations using `EnsembleProblem`s
+Especially when performing stochastic simulations, one often wishes to run a large number of simulations. While it is possible to do so e.g. through a `for` loop, a more convenient way is to create a so-called `EnsembleProblem`. An advantage of these is that they contain a simple interface for modifying a problem across the ensemble, as well as automatic parallelization across CPUs. For a more throughout description, please read [the Parallel Ensemble Simulations section of the DifferentialEquation documentation](https://docs.sciml.ai/DiffEqDocs/stable/features/ensemble/#ensemble). Here, we will give a brief introduction to this feature.
+
+We are using a simple single-component self-activation model (which contains bistability):
+```@example ex1
+rn = @reaction_network begin
+    v0 + hill(X,v,K,n), ∅ --> X
+    deg, X --> ∅
+end
+u0 = [:X => 0.0]
+tspan = (0.0,1000.0)
+p = [:v0 => 0.1, :v => 2.5, :K => 75.0, :n => 2.0, :deg => 0.01];
+sprob = SDEProblem(rn,u0,tspan,p)
+nothing # hide
+```
+we can then use our `SDEProblem` as input to an `EnsembleProblem`:
+```@example ex1
+eprob = EnsembleProblem(sprob)
+```
+next, the `EnsembleProblem` can be used as an input to the `solve()` command. It has the same options as when simulating the `SDEProblem` directly, however, it has an additional argument `trajectories` to determine how many simulations should be performed. 
+```@example ex1
+esol = solve(eprob; trajectories=5)
+```
+This simulation is automatically multithreaded over all available threads. Please read [this documentation](https://docs.sciml.ai/DiffEqDocs/stable/features/ensemble/#EnsembleAlgorithms) for more information on parallelisation alternatives. The ensemble simulations can be plotted using the `plot()` function, which will by default display all trajectories:
+```@example ex1
+plot(esol)
+```
+
+Sometimes when performing a large number of ensemble simulations, the plots get very dense. In these cases, the plotting argument is `linealpha` (which sets trajectory transparency) may be useful:
+```@example ex1
+esol = solve(eprob; trajectories=100)
+plot(esol)
+```
+
+Sometimes, one wishes to perform the same simulation a large number of times, while making minor modifications to the problem each time. This can be done by giving a problem function argument to the `EnsembleProblem`. Let us consider ODE simulations of a simple birth/death process:
+```@example ex1
+rn = @reaction_network begin
+    (b,1.0), ∅ <--> X
+end
+u0 = [:X => 1.0]
+tspan = (0.0,1.0)
+p = [:b => 1.];
+oprob = ODEProblem(rn,u0,tspan,p)
+nothing # hide
+```
+We wish to simulate this model for a large number of values of `b`. We do this by creating a `prob_func` that will make a modification to the problem at the start of each Monte Carlo simulation:
+```@example ex1
+b_values = 1.0:0.1:2.0
+function prob_func(prob,i,repeat)
+    remake(prob; p=[b_values[i]])
```

Comment:
This should use a parameter mapping and not rely on an implicit ordering (even if it is just one state, we don't want to get people in the habit of using implicit orderings).

Reply:
This will likely change as the MTK issues get fixed, hopefully, it will haven vaguely fast. Depending on those, this will change. Happy to keep this open until we know what syntax we can use.

Reply:
Whatever syntax gets accepted there it will still make use of the symbolic for indexing. You can get the symbolic like
```julia
@unpack X = prob.f.sys
remake(prob; p = [X => b_values[i]])
```
I think.

Reply:
Tried this, but the returned problem is identical to the unmodified one. Think this was similar to one of the issues we had previously, I'll check if I can find what went wrong there.

Reply:
My code had a typo and used X instead of b. Does it work with the right symbolic?

Reply:
Right, yeah, I should have caught that.

We use `@unpack` here now, but we still have to use indexing for the callback, as I don't think you can symbolically set index values for integrators...